lib/array.js

```diff
@@ -127,6 +132,30 @@ internals.checkItems = function (items, wasArray, state, options) {
             continue;
         }
 
+        // Ordered
+
```

Comment:
You should just move that block below the exclusions one, exclusions always have priority over everything else.